### ISSUE TITLE
Getting error General error: 2014 with facet()

### ISSUE THREAD

**Issue by pahanini** [yiisoft/yii2-sphinx#27]
Hi,

here is example of the code.

```php
        $query = Item::find();
        $query->addFacets(['category_id']);
        $result = $query->search();
```

This code causes this error:
```
yii\db\Exception: SQLSTATE[HY000]: General error: 2014 Cannot execute queries while other unbuffered queries are active.  Consider using PDOStatement::fetchAll().  Alternatively, if your code is only ever going to run against mysql, you may enable query buffering by setting the PDO::MYSQL_ATTR_USE_BUFFERED_QUERY attribute.
The SQL being executed was: SHOW TABLES
```

If I remove facet() function call from code then search() works fine.

**Comment by pahanini** [comment]
Its because yii tries to call `SHOW TABLES`  while populating active record but at this moment not all the results are read from dataReader.

Here is the diry workaround:
```php
        Item::findOne(1); // <--- It will call SHOW TABLES
        $query = Item::find();
        $query->addFacets(['category_id']);
        $result = $query->search();
```

**Comment by klimov-paul** [comment]
Have you tried to enable `PDO::MYSQL_ATTR_USE_BUFFERED_QUERY`?

**Comment by pahanini** [comment]
Yes, It does not help.

I tried to enable it in this way (config/main.php)
```php
        'sphinx' => [
            'username' => '',
            'password' => '',
            'attributes' => [
                \PDO::MYSQL_ATTR_USE_BUFFERED_QUERY => true
            ],
        ],

```

**Comment by klimov-paul** [comment]
Issue confirmed.

> Yes, It does not help.

`PDO::MYSQL_ATTR_USE_BUFFERED_QUERY` should be set to `PDOStatement` instance - not the `PDO`, but still it can not help: if set the followig error occurs:
```
PDOException: SQLSTATE[IM001]: Driver does not support this function: This driver doesn't support setting attributes
```

**Comment by klimov-paul** [comment]
Issue resolved by commit 6e6b7d5

**Comment by klimov-paul** [comment]
Fix adjusted to be more reliable: 3104802
